pkg/controllers/overrider/clusterresource_controller.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package overrider features controllers to reconcile the override objects.
+package overrider
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	placementv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/utils"
+	"go.goms.io/fleet/pkg/utils/controller"
+	"go.goms.io/fleet/pkg/utils/labels"
+)
+
+// Reconciler reconciles a clusterResourceOverride object.
+type ClusterResourceReconciler struct {
+	Reconciler
+	recorder record.EventRecorder
+}
+
+// Reconcile triggers a single  reconcile round when scheduling policy has changed.
+func (r *ClusterResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
we can probably create a common reconciler for both ro and cro. We can use interface to have different implementations, so that we don't need to duplicate the code. These two controllers are almost the same.

Reply:
we can refactor the code later to see if we can find a clean way.

Reply:
The tricky part is the test, it will be a lot of redundant test too, that's why I stopped at the CRO IT.